Add documentation and plugin notice about the end of the plugin version (#2800)
timber/timber: readme.txt

Helps you create themes faster with sustainable code. With Timber, you write HTML using Twig Templates http://www.upstatement.com/timber/

== Description ==
With the upcoming release of Timber 2, we will not release a 2.0 version and beyond  in plugin form. We advice everyone to switch to the composer based install as soon as possible.

> You will find an extensive list with guides and the reasons why we are not going to release Timber 2 in plugin version anymore.

Reviewer: ```suggestion
You will find an extensive list with guides and the reasons why we are not going to release Timber 2.0 as a plugin anymore.
```

> ### Switching to the composer based version

Reviewer: ```suggestion
### Switching to the Composer based version
```

> * Guide: [How do I switch over from the plugin version to the composer based version of Timber?](https://timber.github.io/docs/getting-started/switch-to-composer/)

Reviewer: ```suggestion
* Guide: [How do I switch over from the plugin version to the Composer based version of Timber?](https://timber.github.io/docs/getting-started/switch-to-composer/)
```

* Backstory: [Why are we dropping support for the plugin in the first place](https://github.com/timber/timber/pull/2005)

> * Github issue: [Road to Timber 2.0](https://github.com/timber/timber/issues/2741)

Reviewer: ```suggestion
* GitHub issue: [Roadmap for Timber 2.0](https://github.com/timber/timber/issues/2741)
```
